plugins/scheme-tpm-enacttrust/main.go

```diff
@@ -101,6 +216,67 @@ func synthKeysFromParts(scope, tenantID string, parts *structpb.Struct) ([]strin
 	return []string{lookupKey.String()}, nil
 }
 
+func parseKey(keyString string) (*ecdsa.PublicKey, error) {
+	buf, err := base64.StdEncoding.DecodeString(keyString)
+	if err != nil {
+		return nil, err
+	}
+
+	key, err := x509.ParsePKIXPublicKey(buf)
+	ret, ok := key.(*ecdsa.PublicKey)
+	if !ok {
+		return nil, fmt.Errorf("could not extract EC public key; got [%T]: %v", key, err)
+	}
+
+	return ret, err
+}
+
+type Token struct {
+	AttestationData *tpm2.AttestationData
+	Raw             []byte
+	Signature       *tpm2.Signature
+}
+
+func (t *Token) Decode(data []byte) error {
+	// The first two bytes are the SIZE of the following TPMS_ATTEST
+	// structure. The following SIZE bytes are the TPMS_ATTEST structure,
+	// the remaining bytes are the signature.
+	if len(data) < 3 {
+		return fmt.Errorf("could not get data size; token too small (%d)", len(data))
+	}
+
+	size := binary.BigEndian.Uint16(data[:2])
+	if len(data) < int(2+size) {
+		return fmt.Errorf("token data appears truncated; expected %d, but got %d bytes",
+			2+size, len(data))
+	}
+
+	var err error
+
+	t.Raw = data[2 : 2+size]
+	t.AttestationData, err = tpm2.DecodeAttestationData(t.Raw)
+	if err != nil {
+		return fmt.Errorf("could not decoded TPMS_ATTEST: %v", err)
+	}
+
+	t.Signature, err = tpm2.DecodeSignature(bytes.NewBuffer(data[2+size:]))
+	if err != nil {
+		return fmt.Errorf("could not decoded TPMT_SIGNATURE: %v", err)
```

Comment:
Did you mean TPMS_SIGNATURE here ?

Reply:
No, TPMT_SIGNATURE is correct. TPMT indicates a Tagged structure (where as TPMS is an untagged, unsized structure).

Reply:
Ok thanks for the clarification! I will resolve the comment now!